gnomad_qc/v4/analyses/grpmax_comps.py

```diff
@@ -0,0 +1,419 @@
+    grpmax_counts: bool = False,
+    non_syn_only: bool = False,
+    eur_filter: bool = False,
+):
+    """
+    Create tables of grpmax stats.
+
+    Tables have a column for each version as well as the difference for each grpmax group.
+
+    :param version_dict: Dictionary of grpmax stats
+    :param data_subset: Data subset
+    :param grpmax_counts: Results calculated using only grpmax counts for diverse genetic ancestry groups
+    :param non_syn_only: Keep only non-synonymous variants
+    :param eur_filter: Filter to variants where eur AF is < threshold while grpmax > threshold
+    """
+    headers = ["genetic ancestry group", "v2", "v4", "difference(v4-v2)"]
+    v2_dict_index = (
+        version_dict["v2"][data_subset]
+        if grpmax_counts and eur_filter
+        else version_dict["v2"]
+    )
+    v4_dict_index = (
+        version_dict["v4"][data_subset]
+        if grpmax_counts and eur_filter
+        else version_dict["v4"]
+    )
+    grps = deepcopy(DIVERSE_GRPS)
+    grps = grps + ["all"]
+
+    if not eur_filter:
+        grps.append("nfe")
+
+    for threshold in AF_THRESHOLDS:
+        table = []
+        for grp in grps:
+            v2_val = v2_dict_index[threshold].get(grp, 0)
+            v4_val = v4_dict_index[threshold].get(grp, 0)
+            diff = v4_val - v2_val
+            if diff < 0:
+                diff = f"\033[91m{diff}\033[0m"
+            else:
+                diff = f"\033[92m{diff}\033[0m"
+            table.append([grp, v2_val, v4_val, diff])
+        logger.info(
+            f"\n{'non-synonymous' if non_syn_only else ''} variant count by genetic "
+            f"ancestry group where the{' grpmax' if grpmax_counts else ''} AF is above"
+            f" {threshold}{' and the ' + str(data_subset) + ' AF is below it' if grpmax_counts and eur_filter else ''}...\n"
+            f"{tabulate(table, headers=headers, tablefmt='fancy_grid')}"
+        )
+
+
+def main(args):
+    """Find grpmax stats for gnomAD v4 and v2."""
+    version_dict = {}
+    grpmax_counts = args.grpmax_counts
+    non_syn_only = args.non_syn_only
+    csq_terms = args.csq_terms
+    eur_filter = args.eur_filter
+    csq_terms = csq_terms if csq_terms else non_syn_only
+
+    for version in ["v4", "v2"]:
+        if version == "v4":
+            ht = release_sites().ht()
+        elif args.use_v2_liftover:
+            ht = hl.read_table(get_gnomad_liftover_data_path("exomes"))
+            vep_ht = vep_context.versions["105"].ht()
+            ht = ht.annotate(vep=vep_ht[ht.key].vep)
+        else:
+            ht = get_gnomad_public_data("exomes")
+
+        if args.test:
+            # Filter to two partitions
+            ht = ht._filter_partitions([0, 1])
+        t_variants = ht.count()
+
+        logger.info(
+            "Total number of variants in %s before filtering: %s", version, t_variants
+        )
+        msg = ""
+        ht = process_consequences(ht, has_polyphen=False)
+
+        if csq_terms:
```

Comment:
this doesn't need an if, right? `csq_terms` should always be defined based on the logic above

Reply:
🔎